### DISCUSSION TITLE
none-ls.nvim problem

### DISCUSSION THREAD

**Discussion by Tianci-King** [ayamir/nvimdots#1202] [Q&A]
Hello, I am a beginner in computer science from China, trying to use the nvim tool. My device is a macOS M1. After following your tutorial and completing all the configurations, I successfully entered nvim. The top right corner prompted that my package will be deprecated, but after reading issue 58 in the link, I still don't know how to adjust formatting/deno_fmt.lua. How can I solve this to avoid this warning tip? 

Here is the tip from nvim：
"［null-Ls］You required a deprecated builtin （formatting/deno_fmt.lua）， which will be removed in March. Please migrate to alternatives: https://github.com/nvimtools/none-ls.nvim/issues/58"

<img width="1384" alt="未命名" src="https://github.com/ayamir/nvimdots/assets/109196852/116aa6b8-0e7f-452c-bdf5-d74753dcfdce">

**Comment by ayamir** [comment]
It means you are using `deno_fmt` which is deprecated builtin formatter of `none-ls.nvim`.
https://github.com/nvimtools/none-ls.nvim/issues/58 notes that you should use `denols` to do format function.
![image](https://github.com/ayamir/nvimdots/assets/61657399/d3dc5640-7b52-476a-b0a9-15c7a32a720f)
So you should stop using `deno_fmt` from `none-ls.nvim` and use `:MasonInstall deno` to using `denols`.

**Reply by Tianci-King** [reply]
sorry, I truly understand your answer, it helps me a lot！Sadly I try  but still do not know how to delete `deno_fmt.lua`  and  install  `deno` , I try to open my `.config` and want to search the `deno_fmt.lua` , but I failed .

how can I stop using deno_fmt from none-ls.nvim and use :MasonInstall deno ? I'd be very grateful if you could tell me how to do it！🥹

here is my folder screenshot：
![1711617316218](https://github.com/ayamir/nvimdots/assets/109196852/11991774-58b6-480b-911e-6ed4b26f6ac5)

Best wish🥰

**Reply by ayamir** [reply]
You can use `:Mason` to check whether `deno` related formatter or lsp is installed? Uninstall it?
